feature_engine/creation/decision_tree.py

```diff
@@ -0,0 +1,328 @@
+            if output_features > len(variables):
+                raise ValueError(
+                    "If output_features is an integer, the value cannot be "
+                    f"greater than the length of variables. Got {output_features} "
+                    "for output_features and {len(variables)} for the length "
+                    "of variables."
+                )
+
+        if isinstance(output_features, list):
+            if (
+                    max(output_features) > len(variables)
+                    or not all(
+                            isinstance(feature, int) for feature in output_features
+                        )
+            ):
+                raise ValueError(
+                    "output_features must be a list solely comprised of integers "
+                    "and the maximum integer cannot be greater than the length of "
+                    f"variables. Got {output_features} for output_features and "
+                )
+
+        if isinstance(output_features, tuple):
+            num_combos = 0
+            for n in range(1, len(variables) + 1):
+                num_combos += len(list(combinations(variables, n)))
+            if (
+                    not all(
+                        isinstance(feature, (str, tuple)) for feature in output_features
+                    )
+                    or len(output_features) > num_combos
+            ):
+                raise ValueError(
+                    "output_features must a tuple solely comprised of tuples and "
+                    f"the maximum number of tuples cannot exceed {num_combos}. Got "
+                    f"{output_features} instead."
+                )
+
+        if not isinstance(regression, bool):
+            raise ValueError(
+                f"regression must be a boolean value. Got {regression} instead."
+            )
+
+        if not isinstance(max_depth, int):
+            raise ValueError(
+                f"max_depth must be an integer. Got {max_depth} instead."
+            )
+
+        super().__init__(missing_value, drop_original)
+        self.variables = variables
+        self.output_features = output_features
+        self.regression = regression
+        self.max_depth = max_depth
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+        The transformer learns the target variable values associated with
+        the user-provided features using a decision tree.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The training input samples. Can be the entire dataframe, not just
+            the variables to transform.
+
+        y: pandas Series or np.array = [n_samples,]
+            The target variable that is used to train the decision tree.
+        """
+        # common checks and attributes
+        # TODO: We don't need to check_variables b/c BaseCreation fit() performs action
+        X = super().fit(X, y)
```

Comment:
In fact, BaseCreation checks for nan so not sure this base class is the best?

Reply:
BaseTransformer does too. I changed the base classes to BaseEstimator and TransformerMixin.